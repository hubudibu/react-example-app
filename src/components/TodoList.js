import React from 'react';
import TodoListItem from './TodoListItem.js';

const TodoList = (props) => (<ul>
  {props.todos.map(todo => <TodoListItem todo={todo} />)}
</ul>);

export default TodoList;
