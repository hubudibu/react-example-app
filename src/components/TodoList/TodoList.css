.TodoList {
  padding: 1em 2em;
  margin: 0;
}
