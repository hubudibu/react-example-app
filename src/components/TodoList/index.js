import React from 'react';
import './style.css';
import TodoListItem from '../TodoListItem';

const TodoList = (props) => (<ul className="TodoList">
  {props.todos && props.todos.map(todo => <TodoListItem todo={todo} />)}
</ul>);

export default TodoList;
