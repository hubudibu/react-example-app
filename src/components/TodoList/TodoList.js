import React from 'react';
import './TodoList.css';
import TodoListItem from './TodoListItem.js';

const TodoList = (props) => (<ul className="TodoList">
  {props.todos.map(todo => <TodoListItem todo={todo} />)}
</ul>);

export default TodoList;
