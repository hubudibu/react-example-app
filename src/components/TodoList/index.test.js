import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from './';

describe('Todo List Item', () => {
  it('renders empty list as in snapshot', () => {
    const wrapper = renderer.create(<TodoList />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders todo item as in snapshot', () => {
    const todos = [
      {
        text: 'Sample todo 1',
        id: 1
      },
      {
        text: 'Sample todo 2',
        id: 2
      }];
    const wrapper = renderer.create(<TodoList todos={todos} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
})
