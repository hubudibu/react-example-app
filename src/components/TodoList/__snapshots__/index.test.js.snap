exports[`Todo List Item renders empty list as in snapshot 1`] = `
<ul
  className="TodoList" />
`;

exports[`Todo List Item renders todo item as in snapshot 1`] = `
<ul
  className="TodoList">
  <li
    className="TodoListItem">
    Sample todo 1
  </li>
  <li
    className="TodoListItem">
    Sample todo 2
  </li>
</ul>
`;
